app/src/componentLibrary/autocompletes/multipleAutocomplete/multipleAutocomplete.jsx

```diff
@@ -201,10 +202,20 @@ export const MultipleAutocomplete = ({
             >
               {({ placement, ref, style, scheduleUpdate }) => {
                 updatePosition = scheduleUpdate;
-                const filteredOptions = options.filter(
-                  (item) =>
-                    value.indexOf(item) < 0 && item.toLowerCase() !== inputValue.toLowerCase(),
-                );
+                const filteredOptions = options.filter((option) => {
+                  const isOptionSelect = value.indexOf(option) !== -1;
+                  const isOptionEqualInputValue = option.toLowerCase() === inputValue.toLowerCase();
+                  const isInputValueMatchAdditionalCreationCondition = getAdditionalCreationCondition(
+                    inputValue,
+                  );
+
+                  return (
+                    !isOptionSelect &&
+                    (duplicateInputValueOption ||
+                      !isOptionEqualInputValue ||
+                      !isInputValueMatchAdditionalCreationCondition)
+                  );
+                });
```

Comment:
If there are no concerns, I suggest to left simple condition here:
```javascript
const filteredOptions = options.filter((item) => value.indexOf(item) < 0);
```
I didn't find any reasons for excluding the options with same value as typed in input.